GitHubTeams.ps1

```diff
@@ -691,219 +1629,328 @@ filter Set-GitHubTeam
+        them individually.
 
     .PARAMETER AccessToken
         If provided, this will be used as the AccessToken for authentication with the
         REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
 
     .INPUTS
         GitHub.Organization
+        GitHub.Project
         GitHub.Team
 
     .OUTPUTS
-        GitHub.Team
+        None
 
     .EXAMPLE
-        Rename-GitHubTeam -OrganizationName PowerShell -TeamName Developers -NewTeamName DeveloperTeam
+        Set-GitHubTeamProjectPermission `
+            -Organization PowerShell `
+            -TeamName Everybody `
+            -Project 4378613
 
-        Renames the 'Developers' GitHub team in the 'PowerShell' organization to be 'DeveloperTeam'.
+        Sets permissions for a project specified by its ID for team 'Everybody' in organization
+        'PowerShell' to the current permissions for the team.
 
     .EXAMPLE
-        $team = Get-GitHubTeam -OrganizationName PowerShell -TeamName Developers
-        $team | Rename-GitHubTeam -NewTeamName 'DeveloperTeam'
-
-        You can also pipe in a GitHub team that was returned from a previous command.
-
-    .NOTES
-        This is a helper/wrapper for Set-GitHubTeam which can also rename a GitHub Team.
+        Set-GitHubTeamProjectPermission `
+            -Organization PowerShell `
+            -TeamName Everybody `
+            -Project 4378613 `
+            -Permission Admin
+
+        Sets permissions for a project specified by its ID for team 'Everybody' in organization
+        'PowerShell' explicitly to the specified permission.
 #>
     [CmdletBinding(
+        SupportsShouldProcess,
         PositionalBinding = $false,
         DefaultParameterSetName = 'TeamSlug')]
-    [OutputType( { $script:GitHubTeamTypeName } )]
     param
     (
         [Parameter(
             Mandatory,
-            ValueFromPipelineByPropertyName,
-            Position = 1)]
+            ValueFromPipelineByPropertyName)]
         [ValidateNotNullOrEmpty()]
-        [string] $OrganizationName,
+        [String] $OrganizationName,
 
         [Parameter(
             Mandatory,
             ValueFromPipelineByPropertyName,
-            Position = 2,
             ParameterSetName='TeamName')]
         [ValidateNotNullOrEmpty()]
-        [string] $TeamName,
+        [String] $TeamName,
 
         [Parameter(
             Mandatory,
             ValueFromPipelineByPropertyName,
             ParameterSetName='TeamSlug')]
-        [ValidateNotNullOrEmpty()]
         [string] $TeamSlug,
 
         [Parameter(
             Mandatory,
-            ValueFromPipelineByPropertyName,
-            Position = 3)]
-        [ValidateNotNullOrEmpty()]
-        [string] $NewTeamName,
+            ValueFromPipelineByPropertyName)]
+        [Alias('ProjectId')]
+        [int64] $Project,
 
-        [switch] $PassThru,
+        [ValidateScript({ -not $_ -or @('Admin', 'Read', 'Write') -contains $_ })]
```

Comment:
You're making this harder on yourself.  You want to use `ValidateSet`.  The added bonus is then you get proper tab completion at the commandline.

```suggestion
        [ValidateSet('Admin', 'Read', 'Write')]
```